FitGoal/FitGoal/Singup&Singin/CustomUIViews/SocialMediaAuthenticationView.swift

```diff
@@ -0,0 +1,129 @@
+        let stack = UIStackView()
+        stack.axis = .horizontal
+        stack.alignment = .center
+        stack.distribution = .fill
+        stack.spacing = 16
+        
+        let leadingLine = UIView()
+        leadingLine.layer.borderColor = #colorLiteral(red: 0.94, green: 0.94, blue: 0.94, alpha: 1).cgColor
+        leadingLine.layer.borderWidth = 1
+        
+        let trailingLine = UIView()
+        trailingLine.layer.borderColor = #colorLiteral(red: 0.94, green: 0.94, blue: 0.94, alpha: 1).cgColor
+        trailingLine.layer.borderWidth = 1
+        
+        let label = UILabel()
+        label.attributedText = "Or".formattedText(
+            font: "Roboto-Light",
+            size: 15,
+            color: #colorLiteral(red: 0.52, green: 0.53, blue: 0.57, alpha: 1),
+            kern: 0
+        )
+        
+        stack.addArrangedSubview(leadingLine)
+        stack.addArrangedSubview(label)
+        stack.addArrangedSubview(trailingLine)
+        
+        leadingLine.widthAnchor.constraint(equalToConstant: 130).isActive = true
+        leadingLine.heightAnchor.constraint(equalToConstant: 2).isActive = true
+        trailingLine.widthAnchor.constraint(equalToConstant: 130).isActive = true
+        trailingLine.heightAnchor.constraint(equalToConstant: 2).isActive = true
+        
+        return stack
+    }()
+    
+    private let socialMediaStack: UIStackView = {
+        let stack = UIStackView()
+        stack.axis = .horizontal
+        stack.alignment = .center
+        stack.spacing = 50
+        stack.distribution = .fill
+        return stack
+    }()
+    
+    private let googleButton: UIButton = {
+        let googleButton = UIButton()
+        googleButton.bounds.size = CGSize(width: 72, height: 72)
+        googleButton.layer.cornerRadius = 40
+        googleButton.contentMode = .scaleAspectFit
+        googleButton.setImage(#imageLiteral(resourceName: "GoogleButton"), for: .normal)
+        googleButton.layer.shadowOffset = CGSize(width: 0, height: 1)
+        googleButton.layer.shadowColor = #colorLiteral(red: 0, green: 0, blue: 0, alpha: 0.01)
+        googleButton.layer.shadowRadius = 5
+        return googleButton
+    }()
+
+    private var facebookButton: UIButton = {
+        let facebookButton = UIButton()
+        facebookButton.bounds.size = CGSize(width: 72, height: 72)
+        facebookButton.layer.cornerRadius = 40
+        facebookButton.contentMode = .scaleAspectFit
+        facebookButton.setImage(#imageLiteral(resourceName: "FacebookButton"), for: .normal)
+        return facebookButton
+    }()
+
+    private var twitterButton: UIButton = {
+        let twitterButton = UIButton()
+        twitterButton.bounds.size = CGSize(width: 72, height: 72)
+        twitterButton.layer.cornerRadius = 40
+        twitterButton.contentMode = .scaleAspectFit
+        twitterButton.setImage(#imageLiteral(resourceName: "TwitterButton"), for: .normal)
+        return twitterButton
+    }()
+    
```

Comment:
These three buttons have almost the same configuration. It would be better if you create a function that builds a base configuration of the button and then you only modify what's particular to each case. For example:

```swift 
private func socialMediaButton() -> UIButton {
    let button = UIButton()
    button.bounds.size = CGSize(width: 72, height: 72)
    button.layer.cornerRadius = 40
    button.contentMode = .scaleAspectFit
    return button
}
```

And then:
```swift
private lazy var facebookButton: UIButton = {
    let facebookButton = socialMediaButton()
    facebookButton.setImage(#imageLiteral(resourceName: "FacebookButton"), for: .normal)
    return facebookButton
}()

private lazy var twitterButton: UIButton = {
    let twitterButton = socialMediaButton()
    twitterButton.setImage(#imageLiteral(resourceName: "TwitterButton"), for: .normal)
    return twitterButton
}()

private lazy var googleButton: UIButton = {
    let googleButton = socialMediaButton()
    googleButton.setImage(#imageLiteral(resourceName: "GoogleButton"), for: .normal)
    googleButton.layer.shadowOffset = CGSize(width: 0, height: 1)
    googleButton.layer.shadowColor = #colorLiteral(red: 0, green: 0, blue: 0, alpha: 0.01)
    googleButton.layer.shadowRadius = 5
    return googleButton
}()
```
Now the code is shorter and all the common logic is in only one place. 
**NOTE**: For this to work you have to mark the fields as `lazy`.